### PR TITLE
ibmcloud: Update security groups and rules

### PR DESCRIPTION
**Pull request by hasueki**
Define more IBM Cloud VPC security groups and rules, following the [documentation](https://docs.openshift.com/container-platform/4.7/installing/installing_platform_agnostic/installing-platform-agnostic.html#installation-network-user-infra_installing-platform-agnostic) for OCP networking requirements. This also updates the set of security groups to attach to the nodes defined in Terraform and Machine/MachineSet CRs.

### CODE REVIEW

File: data/data/ibmcloud/vpc/security-groups.tf [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+}
+
+# Kubernetes default ports
+resource "ibm_is_security_group_rule" "openshift_network_kube_default_ports_inbound" {
+  group     = ibm_is_security_group.openshift_network.id
+  direction = "inbound"
+  remote    = ibm_is_security_group.openshift_network.id
+  tcp {
+    port_min = 10250
+    port_max = 10259
```
Comment (jstuever) [inline data/data/ibmcloud/vpc/security-groups.tf:119]: While port 10250 needs to be open between all hosts, 10257 and 10259 should only be open on the masters. I recommend capping this at 10250 and adding a separate rule for 10257-10259 under the control_plane SG.

**Review by jstuever — CHANGES_REQUESTED** [review]
Can you explain the difference between cluster-wide and openshift-network security groups. They feel redundant and can probably be collapsed into one.

Other than this, a few minor adjustments to tighten security a bit.

File: data/data/ibmcloud/vpc/security-groups.tf [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+  resource_group = var.resource_group_id
+  tags           = var.tags
+  vpc            = ibm_is_vpc.vpc.id
+}
+
+# SSH
+resource "ibm_is_security_group_rule" "cluster_wide_ssh_inbound" {
+  group     = ibm_is_security_group.cluster_wide.id
+  direction = "inbound"
+  remote    = var.public_endpoints ? "0.0.0.0/0" : ibm_is_security_group.cluster_wide.id
```
[the remark this answers is not in the document]
Comment (hasueki) [reply data/data/ibmcloud/vpc/security-groups.tf:24]: Aha ok, thanks for the clarification. I will get this updated.

File: data/data/ibmcloud/vpc/security-groups.tf [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+# Metrics
+resource "ibm_is_security_group_rule" "cluster_wide_metrics_inbound" {
+  group     = ibm_is_security_group.cluster_wide.id
+  direction = "inbound"
+  remote    = ibm_is_security_group.cluster_wide.id
+  tcp {
+    port_min = 1936
+    port_max = 1936
+  }
+}
```
[the remark this answers is not in the document]
Comment (hasueki) [reply data/data/ibmcloud/vpc/security-groups.tf:48]: Ok I was simply following the docs. Will remove this rule.

File: data/data/ibmcloud/vpc/security-groups.tf (hunk 1 above)
Comment (hasueki) [reply data/data/ibmcloud/vpc/security-groups.tf:119]: Got it. Will update.

**Comment by hasueki** [comment]
@jstuever 
> Can you explain the difference between cluster-wide and openshift-network security groups. They feel redundant and can probably be collapsed into one.

There is a limitation in IBM Cloud VPC security groups where only 5 remote rules are configurable per security group. With that, I needed to separate it into 2 security groups and tried to organize it a way that makes the most sense.

**Review by jstuever — APPROVED** [review]

**Pull request merged**